vault_program/src/create_token_metadata.rs

```diff
@@ -1,11 +1,71 @@
-use solana_program::{account_info::AccountInfo, entrypoint::ProgramResult, pubkey::Pubkey};
+use jito_account_traits::AccountDeserialize;
+use jito_jsm_core::loader::{load_signer, load_system_account, load_system_program};
+use jito_vault_core::{loader::load_vault, vault::Vault};
+use solana_program::{
+    account_info::AccountInfo, borsh1::get_instance_packed_len, entrypoint::ProgramResult,
+    program_error::ProgramError, pubkey::Pubkey, rent::Rent, sysvar::Sysvar,
+};
+use spl_pod::optional_keys::OptionalNonZeroPubkey;
+use spl_token_metadata_interface::state::TokenMetadata;
+use spl_type_length_value::state::TlvStateMut;
 
 pub fn process_create_token_metadata(
-    _program_id: &Pubkey,
-    _accounts: &[AccountInfo],
-    _name: String,
-    _symbol: String,
-    _uri: String,
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    name: String,
+    symbol: String,
+    uri: String,
 ) -> ProgramResult {
+    let [metadata_info, vault_info, vault_admin, system_program] = accounts else {
```

Comment:
can you add a payer here too? if the vault_admin is a PDA, it gets kinda weird bc you need to transfer assets to the PDA so it can create the account

Reply:
added `payer` accounts.